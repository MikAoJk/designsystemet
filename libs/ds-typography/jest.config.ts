export default {
  displayName: 'ds-typography',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': ['babel-jest', { presets: ['@nx/react/babel'] }],
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/ds-typography',
};
