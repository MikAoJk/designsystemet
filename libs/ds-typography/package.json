{
  "name": "@skatteetaten/ds-typography",
  "version": "0.0.1",
  "groupId": "no.skatteetaten.aurora",
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "react": "^18"
  }
}
