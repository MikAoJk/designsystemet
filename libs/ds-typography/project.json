{
  "name": "ds-typography",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-typography/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "build": {
      "executor": "@nrwl/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-typography",
        "tsConfig": "libs/ds-typography/tsconfig.lib.json",
        "project": "libs/ds-typography/package.json",
        "entryFile": "libs/ds-typography/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-typography/rollup.config.js"],
        "compiler": "babel",
        "assets": [
          {
            "glob": "libs/ds-typography/README.md",
            "input": ".",
            "output": "."
          }
        ]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/ds-typography/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/ds-typography/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-typography/**/*.css",
          "libs/ds-typography/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      }
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-typography"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish {args.switches}",
        "cwd": "dist/libs/ds-typography"
      }
    }
  },
  "implicitDependencies": ["!ds-dev-config"]
}
