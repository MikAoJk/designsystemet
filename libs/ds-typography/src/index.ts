export * from './Blockquote/Blockquote';
export * from './Blockquote/Blockquote.types';
export * from './Paragraph/Paragraph';
export * from './Paragraph/Paragraph.types';
