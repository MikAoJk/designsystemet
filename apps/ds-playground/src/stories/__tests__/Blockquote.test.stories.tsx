import { Blockquote, BlockquoteProps } from '@skatteetaten/ds-typography';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { ElementHandle, ScreenshotOptions } from 'puppeteer';

const wrapper = '[data-test-block]';
const screenShotOptions: ScreenshotOptions = {
  fullPage: true,
  encoding: 'base64',
};

export default {
  component: Blockquote,
  title: 'Tests / Blockquote',
} as ComponentMeta<typeof Blockquote>;

const defaultArgs: BlockquoteProps = {
  children:
    'Lorem ipsum dolor sit amet. Alle som har laget en nettside, trengt litt fylltekst eller bare surfet rundt på nettet har antageligvis sett disse ordene, ' +
    'etterfulgt av en tilsynelatende eviglang tekst fylt med latinske liksomsetninger.',
};
const Template: ComponentStory<typeof Blockquote> = (args) => (
  <div data-test-block>
    <Blockquote {...args} />
  </div>
);

// Når Blockquote har en ref, så får dom elementet ref forwarded
export const WithRef = Template.bind({});
WithRef.storyName = 'With Ref (FA1)';
WithRef.args = {
  ...defaultArgs,
  ref: (instance: HTMLQuoteElement | null): void => {
    if (instance) {
      instance.id = 'dummyIdForwardedFromRef';
    }
  },
};
WithRef.parameters = {
  async puppeteerTest(page: ElementHandle): Promise<void> {
    const refId = await page.$eval(`${wrapper} > blockquote`, (el) => el.id);
    expect(refId).toBe('dummyIdForwardedFromRef');

    const innerHtml = await page.$eval(wrapper, (el) => el.innerHTML);
    expect(innerHtml).toMatchSnapshot();
  },
};

// Når Blockquote har en id, så har elementet id'en satt
export const WithId = Template.bind({});
WithId.storyName = 'With Id (FA2)';
WithId.args = {
  ...defaultArgs,
  id: 'BlockquoteId',
};
WithId.parameters = {
  async puppeteerTest(page: ElementHandle): Promise<void> {
    const id = await page.$eval(`${wrapper} > blockquote`, (el) =>
      el.getAttribute('id')
    );
    expect(id).toBe('BlockquoteId');

    const innerHtml = await page.$eval(wrapper, (el) => el.innerHTML);
    expect(innerHtml).toMatchSnapshot();
  },
};

// Når Blockquote har en custom CSS, så vises custom stil
export const WithCustomCss = Template.bind({});
WithCustomCss.storyName = 'With Custom CSS (FA3)';
WithCustomCss.args = {
  ...defaultArgs,
  className: 'dummyClassname',
};
WithCustomCss.argTypes = {
  className: {
    control: 'select',
    options: ['', 'dummyClassname'],
  },
};
WithCustomCss.parameters = {
  async puppeteerTest(page: ElementHandle): Promise<void> {
    const classNameAttribute = await page.$eval(
      `${wrapper} > blockquote`,
      (el) => el.getAttribute('class')
    );
    expect(classNameAttribute).toContain('dummyClassname');

    const innerHtml = await page.$eval(wrapper, (el) => el.innerHTML);
    expect(innerHtml).toMatchSnapshot();

    const image = await page.screenshot(screenShotOptions);
    expect(image).toMatchImageSnapshot();
  },
};

// Når Blockquote har dataTestId, så har elementet data-testid satt
export const WithDataTestId = Template.bind({});
WithDataTestId.storyName = 'With DataTestid (FA4)';
WithDataTestId.args = {
  ...defaultArgs,
  'data-testid': 'BlockquoteID',
};
WithDataTestId.parameters = {
  async puppeteerTest(page: ElementHandle): Promise<void> {
    const dataTestId = await page.$eval(`${wrapper} > blockquote`, (el) =>
      el.getAttribute('data-testid')
    );
    expect(dataTestId).toBe('BlockquoteID');

    const innerHtml = await page.$eval(wrapper, (el) => el.innerHTML);
    expect(innerHtml).toMatchSnapshot();
  },
};

// Når Blockquote har en lang, så har elementet lang satt
export const WithLang = Template.bind({});
WithLang.storyName = 'With Lang (FA5)';
WithLang.args = {
  ...defaultArgs,
  lang: 'nb',
};
WithLang.parameters = {
  async puppeteerTest(page: ElementHandle): Promise<void> {
    const langAttribute = await page.$eval(`${wrapper} > blockquote`, (el) =>
      el.getAttribute('lang')
    );
    expect(langAttribute).toBe('nb');

    const innerHtml = await page.$eval(wrapper, (el) => el.innerHTML);
    expect(innerHtml).toMatchSnapshot();
  },
};

// Når Blockquote instansieres, får den riktige default-verdier
export const Defaults = Template.bind({});
Defaults.storyName = 'Defaults (A1, B1 - 1 av 2)';
Defaults.args = {
  ...defaultArgs,
};
Defaults.parameters = {
  async puppeteerTest(page: ElementHandle): Promise<void> {
    const innerHtml = await page.$eval(wrapper, (el) => el.innerHTML);
    const image = await page.screenshot(screenShotOptions);

    expect(innerHtml).toMatchSnapshot();
    expect(image).toMatchImageSnapshot();
  },
};

const TemplateWithMarkup: ComponentStory<typeof Blockquote> = (args) => (
  <div data-test-block>
    <Blockquote {...args}>
      <div>
        {'Manowar Manowar living on the road '}
        <a href={'https://en.wikipedia.org/wiki/Manowar'}>{'lenke'}</a>
      </div>
      <div>
        {"When we're on "}
        <strong>{'strong '}</strong>
        {'explode '}
        <em>{'italic em '}</em>
      </div>
      <div>
        {"We don't attract "}
        <code>{'code wimps '}</code>
        {"'cause we're too "}
        <mark>{'mark'}</mark>
      </div>
      <div>
        {'Just '}
        <i>{'italic i '}</i>
        {'true '}
        <small>{'small '}</small>
        {"people that's Manowar's "}
        <b>{'b'}</b>
      </div>
    </Blockquote>
  </div>
);

// Når Blockquote instansieres med markup, får markup riktig styling
export const WithMarkup = TemplateWithMarkup.bind({});
WithMarkup.storyName = 'With Markup (A2, B1 - 2 av 2)';
WithMarkup.parameters = {
  async puppeteerTest(page: ElementHandle): Promise<void> {
    const innerHtml = await page.$eval(wrapper, (el) => el.innerHTML);
    const image = await page.screenshot(screenShotOptions);

    expect(innerHtml).toMatchSnapshot();
    expect(image).toMatchImageSnapshot();
  },
};

const TemplateWithTwoBlockquotes: ComponentStory<typeof Blockquote> = (
  args
) => (
  <div data-test-block>
    <Blockquote {...args} />
    <Blockquote {...args} />
  </div>
);

// Når Blockquote har spacing, så får elementet en margin under sitatet
export const WithSpacing = TemplateWithTwoBlockquotes.bind({});
WithSpacing.storyName = 'With Spacing (A3)';
WithSpacing.args = {
  ...defaultArgs,
  hasSpacing: true,
};
WithSpacing.parameters = {
  async puppeteerTest(page: ElementHandle): Promise<void> {
    const innerHtml = await page.$eval(wrapper, (el) => el.innerHTML);
    const image = await page.screenshot(screenShotOptions);

    expect(innerHtml).toMatchSnapshot();
    expect(image).toMatchImageSnapshot();
  },
};
