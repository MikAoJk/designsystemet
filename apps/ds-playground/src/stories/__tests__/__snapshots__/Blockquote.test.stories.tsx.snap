// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Blockquote Defaults smoke-test 1`] = `
<div data-test-block="true">
  <blockquote class="Blockquote_blockquote__i5HzN  ">
    Lorem ipsum dolor sit amet. Alle som har laget en nettside, trengt litt fylltekst eller bare surfet rundt på nettet har antageligvis sett disse ordene, etterfulgt av en tilsynelatende eviglang tekst fylt med latinske liksomsetninger.
  </blockquote>
</div>
`;

exports[`Tester/Blockquote WithCustomCss smoke-test 1`] = `
<div data-test-block="true">
  <blockquote class="Blockquote_blockquote__i5HzN  dummyClassname">
    Lorem ipsum dolor sit amet. Alle som har laget en nettside, trengt litt fylltekst eller bare surfet rundt på nettet har antageligvis sett disse ordene, etterfulgt av en tilsynelatende eviglang tekst fylt med latinske liksomsetninger.
  </blockquote>
</div>
`;

exports[`Tester/Blockquote WithDataTestid smoke-test 1`] = `
<div data-test-block="true">
  <blockquote class="Blockquote_blockquote__i5HzN  "
              data-testid="BlockquoteID"
  >
    Lorem ipsum dolor sit amet. Alle som har laget en nettside, trengt litt fylltekst eller bare surfet rundt på nettet har antageligvis sett disse ordene, etterfulgt av en tilsynelatende eviglang tekst fylt med latinske liksomsetninger.
  </blockquote>
</div>
`;

exports[`Tester/Blockquote WithId smoke-test 1`] = `
<div data-test-block="true">
  <blockquote id="BlockquoteId"
              class="Blockquote_blockquote__i5HzN  "
  >
    Lorem ipsum dolor sit amet. Alle som har laget en nettside, trengt litt fylltekst eller bare surfet rundt på nettet har antageligvis sett disse ordene, etterfulgt av en tilsynelatende eviglang tekst fylt med latinske liksomsetninger.
  </blockquote>
</div>
`;

exports[`Tester/Blockquote WithLang smoke-test 1`] = `
<div data-test-block="true">
  <blockquote class="Blockquote_blockquote__i5HzN  "
              lang="nb"
  >
    Lorem ipsum dolor sit amet. Alle som har laget en nettside, trengt litt fylltekst eller bare surfet rundt på nettet har antageligvis sett disse ordene, etterfulgt av en tilsynelatende eviglang tekst fylt med latinske liksomsetninger.
  </blockquote>
</div>
`;

exports[`Tester/Blockquote WithMarkup smoke-test 1`] = `
<div data-test-block="true">
  <blockquote class="Blockquote_blockquote__i5HzN  ">
    <div>
      Manowar Manowar living on the road
      <a href="https://en.wikipedia.org/wiki/Manowar">
        lenke
      </a>
    </div>
    <div>
      When we're on
      <strong>
        strong
      </strong>
      explode
      <em>
        italic em
      </em>
    </div>
    <div>
      We don't attract
      <code>
        code wimps
      </code>
      'cause we're too
      <mark>
        mark
      </mark>
    </div>
    <div>
      Just
      <i>
        italic i
      </i>
      true
      <small>
        small
      </small>
      people that's Manowar's
      <b>
        b
      </b>
    </div>
  </blockquote>
</div>
`;

exports[`Tester/Blockquote WithRef smoke-test 1`] = `
<div data-test-block="true">
  <blockquote class="Blockquote_blockquote__i5HzN  "
              id="dummyIdForwardedFromRef"
  >
    Lorem ipsum dolor sit amet. Alle som har laget en nettside, trengt litt fylltekst eller bare surfet rundt på nettet har antageligvis sett disse ordene, etterfulgt av en tilsynelatende eviglang tekst fylt med latinske liksomsetninger.
  </blockquote>
</div>
`;

exports[`Tester/Blockquote WithSpacing smoke-test 1`] = `
<div data-test-block="true">
  <blockquote class="Blockquote_blockquote__i5HzN Blockquote_blockquote_hasSpacing__uLqf9 ">
    Lorem ipsum dolor sit amet. Alle som har laget en nettside, trengt litt fylltekst eller bare surfet rundt på nettet har antageligvis sett disse ordene, etterfulgt av en tilsynelatende eviglang tekst fylt med latinske liksomsetninger.
  </blockquote>
  <blockquote class="Blockquote_blockquote__i5HzN Blockquote_blockquote_hasSpacing__uLqf9 ">
    Lorem ipsum dolor sit amet. Alle som har laget en nettside, trengt litt fylltekst eller bare surfet rundt på nettet har antageligvis sett disse ordene, etterfulgt av en tilsynelatende eviglang tekst fylt med latinske liksomsetninger.
  </blockquote>
</div>
`;
